tests/test_environment_indexentries.py

```diff
@@ -126,3 +126,21 @@ def test_create_index_by_key(app):
     assert index[0] == ('D', [('docutils', [[('main', '#term-docutils')], [], None])])
     assert index[1] == ('P', [('Python', [[('main', '#term-python')], [], None])])
     assert index[2] == ('ス', [('スフィンクス', [[('main', '#term-0')], [], 'ス'])])
+
+
+@pytest.mark.sphinx('dummy', freshenv=True)
+def test_create_index_with_name(app):
+    text = (".. index:: single: docutils\n   :name: ref1\n"
+            ".. index:: see: Python; interpreter\n   :name: ref2\n")
+    restructuredtext.parse(app, text)
+    index = IndexEntries(app.env).create_index(app.builder)
+
+    # check index is created correctly
+    assert len(index) == 2
+    assert index[0] == ('D', [('docutils', [[('', '#index-0')], [], None])])
+    assert index[1] == ('P', [('Python', [[], [('see interpreter', [])], None])])
+
+    # check the reference labels are created correctly
+    labels = app.env.domaindata['std']['anonlabels']
```

Comment:
Please use this code instead. It makes test code robust because it does not touch inside data structure directly.
```
std = app.env.get_domain('std')
assert std.anonlabels['ref1'] == (...)
```
```

Reply:
Done